### PR TITLE
 mixxx: Update to 2.4.0

### PR DESCRIPTION
**Pull request by malfisya**
**Summary**
 
- Full changelog can be found [here](https://raw.githubusercontent.com/mixxxdj/mixxx/2.4.0/CHANGELOG.md)
- Resolves getsolus/packages#1678

**Test Plan**

Launch the app

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

File: packages/m/microsoft-gsl/package.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    - https://github.com/microsoft/GSL/archive/refs/tags/v4.0.0.tar.gz : f0e32cb10654fea91ad56bde89170d78cfbf4363ee0b01d8f097de2ba49f6ce9
+homepage   : https://github.com/microsoft/GSL
+license    : MIT
+component  : programming.library
+summary    : Guidelines Support Library
+description: |
+    The Guidelines Support Library (GSL) contains functions and types that are suggested for use by the C++ Core Guidelines maintained by the Standard C++ Foundation
+libsplit   : no
+builddeps  :
+    - gtest
```
Comment (davidjharder) [inline packages/m/microsoft-gsl/package.yml:14]: Just double checking: This is a builddep, not a checkdep, correct?

Comment (malfisya) [reply packages/m/microsoft-gsl/package.yml:14]: I think so. There is no `check` step in the recipe.

Comment (malfisya) [reply packages/m/microsoft-gsl/package.yml:14]: Not sure, but it won't build without it

**Review by davidjharder — APPROVED** [review]
Looks good, thanks for the update!

**Pull request merged**